packages/playwright-test/src/common/configLoader.ts

```diff
@@ -27,9 +27,52 @@ import { addToCompilationCache } from '../transform/compilationCache';
 import { initializeEsmLoader } from './esmLoaderHost';
 
 const kDefineConfigWasUsed = Symbol('defineConfigWasUsed');
-export const defineConfig = (config: any) => {
-  config[kDefineConfigWasUsed] = true;
-  return config;
+export const defineConfig = (...configs: any[]) => {
+  let result: any = {};
+  for (const config of configs) {
+    result = {
+      ...result,
+      ...config,
+      expect: {
+        ...result.expect,
+        ...config.expect,
+      },
+      use: {
+        ...result.use,
+        ...config.use,
+      },
+      webServer: [
+        ...(Array.isArray(result.webServer) ? result.webServer : (result.webServer ? [result.webServer] : [])),
+        ...(Array.isArray(config.webServer) ? config.webServer : (config.webServer ? [config.webServer] : [])),
+      ]
+    };
+
+    const projectOverrides = new Map<string, any>();
+    for (const project of config.projects || [])
+      projectOverrides.set(project.name, project);
+
+    const projects = [];
+    for (const project of result.projects || []) {
+      const projectOverride = projectOverrides.get(project.name);
+      if (projectOverride) {
+        projects.push({
+          ...project,
+          ...projectOverride,
+          use: {
+            ...project.use,
+            ...projectOverride.use,
+          }
+        });
+        projectOverrides.delete(project.name);
+      } else {
+        projects.push(project);
+      }
+    }
+    projects.push(...projectOverrides.values());
```

Comment:
This will remove projects with duplicate names from the second config:

```ts
defineConfig({}, {
  projects: [{ testMatch: 'foo' }, { testMatch: 'bar' }],
});
```